backend/dataall/modules/s3_datasets/services/dataset_location_service.py

```diff
@@ -135,3 +134,42 @@ def _delete_dataset_folder_read_permission(session, dataset: S3Dataset, location
         }
         for group in permission_group:
             ResourcePolicyService.delete_resource_policy(session=session, group=group, resource_uri=location_uri)
+
+    @staticmethod
+    def get_folder_restricted_information(folder: DatasetStorageLocation):
+        try:
+            context = get_context()
+            with context.db_engine.scoped_session() as session:
+                ResourcePolicyService.check_user_resource_permission(
+                    session=session,
+                    username=context.username,
+                    groups=context.groups,
+                    resource_uri=folder.locationUri,
+                    permission_name=GET_DATASET_FOLDER,
+                )
+                dataset = DatasetRepository.get_dataset_by_uri(session, folder.datasetUri)
+                return {
```

Comment:
What happens if we return the entire object? Would GQL fail because of the extra fields or will only return those that have mappings?
If that's not possible then maybe something like 
`return {field.name: getattr(dataset, field.name, None) for field in DatasetRestrictedInformation.fields}`
 would work better as we won't have to maintain those mappings manually. You can also export it to global method and reuse it an all places

Reply:
You are right. Let me change it a bit